lib/pwa.js

```diff
@@ -312,82 +342,59 @@ exports.fetchManifest = function(pwa) {
-
-          // Validates the manifest or, reject and return errors in an array
-          const res = libManifest.validateManifest(
-            manifest.raw, pwa.manifestUrl, pwa.absoluteStartUrl);
-          if (res.length !== 0) {
-            return Promise.reject('Error while validating the manifest: ' + res);
-          }
-          pwa.manifest = manifest;
-
-          return this.savePwa(pwa)
-            .then(savedPwa => {
-              return {
-                pwa: savedPwa,
-                created: !existingPwa
-              };
-            });
-        })
-        .then(result => {
-          // finish up in the background
-          Promise.all([
-            this.updatePwaMetadataDescription(result.pwa),
-            this.updatePwaIcon(result.pwa),
-            this.updatePwaLighthouseInfo(result.pwa)])
-          .then(_ => {
-            this.savePwa(result.pwa)
-              .then(() => {
-                if (result.created) {
-                  this.sendNewAppNotification(result.pwa.id);
-                }
-              });
-          });
-          return result.pwa;
-        });
+exports.updatePwaManifest = function(pwa) {
+  return libPwa.fetchManifest(pwa)
+  .then(manifest => {
+    return libPwa.findByManifestUrl(pwa.manifestUrl)
+    .then(existingPwa => {
+      if (existingPwa) {
+        pwa = existingPwa;
+        pwa.updated = new Date();
+      }
+      const validationErrors = libPwa.validateManifest(pwa, manifest);
+      if (validationErrors.length > 0) {
+        return Promise.reject('Error while validating the manifest: ' + validationErrors);
+      }
+      pwa.manifest = manifest;
+      return pwa;
     });
+  });
 };
 
-exports.sendNewAppNotification = function(pwaid) {
-  console.log('Sending Notification for ', pwaid);
-  return this.find(pwaid)
-    .then(pwa => {
-      if (!pwa) {
-        return Promise.reject('PWA not found. ' + pwaid);
-      }
-      const clickAction = config.get('CANONICAL_ROOT') + 'pwas/' + pwa.id;
-      return notificationsLib.sendPush('new-apps', {
-        title: pwa.name + ' added to PWA Directory',
-        body: pwa.description || '',
-        icon: pwa.iconUrl64 || '',
-        click_action: clickAction // eslint-disable-line camelcase
-      });
-    })
-    .catch(err => {
-      console.error('Error sending Push:', err);
+/**
+ * Validate PWA's Manifest.
+ *
+ * @param {Pwa} Pwa to validate
+ * @param {Manifest} Manifest to validate
+ * @returns {errors[]} Return errors in an array
+ */
+exports.validateManifest = function(pwa, manifest) {
+  return libManifest.validateManifest(
+    manifest.raw, pwa.manifestUrl, pwa.absoluteStartUrl);
+};
+
+/**
+ * Sends a push notification for new PWAs using Firebase Cloud Messaging.
+ *
+ * @param {Pwa} pwa to send the notification for
+ * @return {Promise<JSON>} the result of the push notification
+ */
+exports.sendNewAppNotification = function(pwa) {
+  if (pwa.isNew()) {
```

Comment:
nit: I prefer exit early in this case:

```
if (!pwa.isNew()) {
  return;
}
```

it makes the flow easier to follow - but that's purely a matter of taste and feel free to ignore